server/etcdserver/util_test.go

```diff
@@ -110,3 +112,54 @@ type testStringerFunc func() string
 func (s testStringerFunc) String() string {
 	return s()
 }
+
+func TestIsTooManyRequest(t *testing.T) {
+	tests := []struct {
+		ci     uint64
+		ai     uint64
+		expect bool
+		req    *pb.InternalRaftRequest
+		desc   string
+	}{
+		{
+			ci:     1 + maxGapBetweenApplyAndCommitIndex,
+			ai:     1,
+			expect: false,
+			req:    nil,
+			desc:   "Test nil InternalRaftRequest",
+		},
+		{
+			ci:     1 + maxGapBetweenApplyAndCommitIndex,
+			ai:     1,
+			expect: false,
+			req:    &pb.InternalRaftRequest{},
+			desc:   "Test non-critical request and gap is not larger than maxGapBetweenApplyAndCommitIndex",
+		},
+		{
+			ci:     1 + maxGapBetweenApplyAndCommitIndex + 1,
+			ai:     1,
+			expect: true,
+			req:    &pb.InternalRaftRequest{},
+			desc:   "Test non-critical request and gap is larger than maxGapBetweenApplyAndCommitIndex",
+		},
+		{
+			ci:     1 + maxGapBetweenApplyAndCommitIndex + 1,
+			ai:     1,
+			expect: false,
+			req:    &pb.InternalRaftRequest{LeaseRevoke: &pb.LeaseRevokeRequest{}},
+			desc:   "Test critical request and gap is larger than maxGapBetweenApplyAndCommitIndex",
+		},
+		{
+			ci:     1 + maxGapBetweenApplyAndCommitIndex + 1 + maxGapBetweenApplyAndCommitIndex/10,
+			ai:     1,
+			expect: true,
+			req:    &pb.InternalRaftRequest{LeaseRevoke: &pb.LeaseRevokeRequest{}},
+			desc:   "Test critical request and gap is larger than 110% maxGapBetweenApplyAndCommitIndex",
+		},
+	}
+
+	for _, test := range tests {
+		t.Log(test.desc)
+		assert.Equal(t, test.expect, isTooManyRequest(test.ai, test.ci, test.req))
+	}
```

Comment:
pls use golang sub test
```suggestion
	for _, tc := range tests {
		t.Run(tc.name, func(t *testing.T) {
			assert.Equal(t, test.expect, isTooManyRequest(test.ai, test.ci, test.req))
		}
	}
```